build: add mergify rule for backport PRs in release 3.7 branch

humblec opened the pull request:
- [x] Add rule for backporting to release 3.7 branch with
label `backport-to-release-v3.7`
- [x] Remove rule for backporting to release 3.5 branch with 
label `backport-to-release-v3.5`

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
build: add mergify rule for backport PRs in release 3.7 branch

This commit add rule for backporting to release 3.7 branch with
label backport-to-release-v3.7

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

nixpanic reviewed (changes requested):
probably good to remove the rules for v3.5 now too?

humblec committed:
build: remove release 3.5 backport rule

considering we have 3.7 release in place, this commit remove
the release 3.5 backporting based on the labels in the repo.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
> probably good to remove the rules for v3.5 now too?

sure, make sense.. updated the PR with the removal

humblec commented:
@nixpanic can you please take a look at the new version ?

humblec commented:
@Mergifyio refresh
